src/Uno.Extensions.Navigation.UI/Navigators/NavigationViewNavigator.cs

```diff
@@ -71,13 +71,24 @@ private object[] NavigationMenuItems
 				return Array.Empty<object>();
 			}
 
-
-			if (Control.MenuItemsSource is IEnumerable items)
+			if (Control.MenuItemsSource is IEnumerable ||
+				Control.FooterMenuItemsSource is IEnumerable)
 			{
-				return items.OfType<object>().ToArray();
+				var menuItemsSource = Control.MenuItemsSource is IEnumerable<NavigationViewItem> itemsSource ?
+					itemsSource.ToArray() : Array.Empty<NavigationViewItem>();
+				var footerMenuItemsSource = Control.FooterMenuItemsSource is IEnumerable<NavigationViewItem> footerItemsSource ?
+					footerItemsSource.ToArray() : Array.Empty<NavigationViewItem>();
+
+				var allitems = menuItemsSource.Concat(footerMenuItemsSource).ToArray();
+
+				return allitems;
 			}
 
-			return Control.MenuItems.ToArray();
+			var menuItems = Control.MenuItems.OfType<NavigationViewItem>();
+			var footerMenuItems = Control.FooterMenuItems.OfType<NavigationViewItem>();
```

Comment:
I had to use the `NavigationViewItem` type, otherwise, when merging the two lists we would end up with a list inside the other. 

![object](https://github.com/unoplatform/uno.extensions/assets/114886335/bb67bf5f-96fa-4ae7-bad2-7b35768caae3)

Alternatively we could use a foreach and not use `NavigationViewItem`.

Reply:
typically this happen when the two element you are concat-ing arent the same type
and you somehow have an overload for `Concat<T>(this IEnumerable<T>, T)` where `T` was `object`
which shouldve been named `Append` to avoid the confusion